lntest/harness.go

```diff
@@ -0,0 +1,828 @@
+	"github.com/roasbeef/btcd/integration/rpctest"
+	"github.com/roasbeef/btcd/rpcclient"
+	"github.com/roasbeef/btcd/txscript"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+)
+
+// NetworkHarness is an integration testing harness for the lightning network.
+// The harness by default is created with two active nodes on the network:
+// Alice and Bob.
+type NetworkHarness struct {
+	rpcConfig rpcclient.ConnConfig
+	netParams *chaincfg.Params
+	Miner     *rpctest.Harness
+
+	activeNodes map[int]*HarnessNode
+
+	// Alice and Bob are the initial seeder nodes that are automatically
+	// created to be the initial participants of the test network.
+	Alice *HarnessNode
+	Bob   *HarnessNode
+
+	bitcoinWatchRequests chan *txWatchRequest
+
+	// Channel for transmitting stderr output from failed lightning node
+	// to main process.
+	lndErrorChan chan error
+
+	quit chan struct{}
+
+	mtx sync.Mutex
+}
+
+// NewNetworkHarness creates a new network test harness. This initializes alice
+// and bob nodes given an already running instance of btcd's rpctest harness and
+// extra command line flags, which should be formatted properly - "--arg=value".
+//
+// TODO(roasbeef): add option to use golang's build library to a binary of the
+// current repo. This'll save developers from having to manually `go install`
+// within the repo each time before changes
+func NewNetworkHarness(r *rpctest.Harness) (*NetworkHarness, error) {
+	return &NetworkHarness{
+		activeNodes:          make(map[int]*HarnessNode),
+		bitcoinWatchRequests: make(chan *txWatchRequest),
+		lndErrorChan:         make(chan error),
+		netParams:            r.ActiveNet,
+		Miner:                r,
+		rpcConfig:            r.RPCConfig(),
+		quit:                 make(chan struct{}),
+	}, nil
+}
+
+// ProcessErrors returns a channel used for reporting any fatal process errors.
+// If any of the active nodes within the harness' test network incur a fatal
+// error, that error is sent over this channel.
+func (n *NetworkHarness) ProcessErrors() <-chan error {
+	return n.lndErrorChan
+}
+
+// fakeLogger is a fake grpclog.Logger implementation. This is used to stop
+// grpc's logger from printing directly to stdout.
+type fakeLogger struct{}
+
+func (f *fakeLogger) Fatal(args ...interface{})                 {}
+func (f *fakeLogger) Fatalf(format string, args ...interface{}) {}
+func (f *fakeLogger) Fatalln(args ...interface{})               {}
+func (f *fakeLogger) Print(args ...interface{})                 {}
+func (f *fakeLogger) Printf(format string, args ...interface{}) {}
+func (f *fakeLogger) Println(args ...interface{})               {}
+
+// SetUp starts the initial seeder nodes within the test harness. The initial
+// node's wallets will be funded wallets with ten 1 BTC outputs each. Finally
+// rpc clients capable of communicating with the initial seeder nodes are
+// created.
+func (n *NetworkHarness) SetUp(seenTxns <-chan *chainhash.Hash,
```

Comment:
`godoc` comment should document the semantics of `lndArgs`. Based on my earlier comment, I don't think the `seenTxns` arg is needed at all.